e3db/src/main/java/com/tozny/e3db/Realm.kt

```diff
@@ -82,8 +80,44 @@ class Realm @JvmOverloads constructor(realmName: String?, appName: String?, brok
               JacksonConverterFactory.create(mapper)
           )
         }
+
     identityClient = retrofitBuilder.build().create()
 
+    // All network calls are done on background threads
+    // This is finicky, and does not guarantee that realmInfo will be populated before it is used?
```

Comment:
since we know it's finicky can we add some retry logic into the sdk to retry a default and configurable number of times (i.e. I could see due to the increased finickiness of flutter to java wanting to set the reties to a higher number than for normal android app development

Reply:
My understanding is that this should only be finicky because the value has not yet returned, not that the return fails. @Mrhea we should find a way to wait for that to be finished, and if it errored than throw and if it succeeds than you can get the value. Kotlin should have support for this type of concurrency though I don't know it off the top of my head. Needing to do this in a background thread is a bit of a bummer because we can't just do the init synchronously

Reply:
^ Is the correct understanding, finicky because value has not yet returned. 

I will go digging for that, sounds similar to what we just did in Swift for Flutter (a `DispatchGroup` that runs background tasks, and you can wait until all tasks in group are finished)

Reply:
I put the call to public realm info into a coroutine; this is my first attempt at Kotlin coroutines, but it looks like it works and my tests are passing. 

It blocks the caller thread until the coroutine finishes with `job.join()`, so we can use `this.realmInfo.domain` as part of broker target url now

Expect the naming of `job` to change as it's not really meaningful to anybody reading it I think